### ISSUE TITLE
Do not report missing "introspection" block in non-Schema modules, while still report it on the Schema module as expected.

### ISSUE THREAD

**Issue by StylianosGakis** [apollographql/apollo-kotlin#6058]
### Version

4.0.0-rc01

### Summary

![image](https://github.com/user-attachments/assets/01512c59-3095-4dca-a188-e3c09f661435)

I basically only have the introspection in the "schema module", and all other modules that depend on it just have more or less this
```kotlin
apollo {
  service("octopus") {
    packageName.set("octopus")
    dependsOn(projects.ourSchemaModule, bidirectional = true)
  }
}
```
in their gradle configuration along with applying the apollo gradle plugin.

Optimally this would not show in the modules which are only using apollo-kotlin, but are not defining a schema themselves.

Initial discussion: https://slack-chats.kotlinlang.org/t/22707737/this-warning-pic-comes-from-the-gradle-plugin-right-i-basica#fdb92517-43ef-4d6d-9b17-d8a2ff58794d

### Steps to reproduce the behavior

Have more than one module which uses apollo-kotlin, where one is the schema module, and notice that all non-schema modules get this warning even though they do not want to also define the introspection there.

**Comment by BoD** [comment]
Thanks for reporting!

**Comment by martinbonnin** [comment]
Released in 4.0.0
